### PR TITLE
fix: allows hard delete for feedbacks

### LINKED ISSUE #6153: No hard delete for feedback
**Describe the bug**

Currently there is hard delete for feedback. When a user sets his/her rating to 0 again, the feedback of the user should be removed completely from the db.
This issue needs to be solved to merge https://github.com/fossasia/open-event-frontend/pull/3198

**Expected behavior**

When the user clears his/her rating (i.e. sets it to 0 again), the feedback must be removed.

**Additional context**

On it

### PR DESCRIPTION
**Pull request by shreyanshdwivedi**
Fixes #6153 

#### Short description of what this resolves:
Currently there is hard delete for feedback. 

#### Changes proposed in this pull request:
- Hard deletes the feedback if deleted_at is not NULL
- Changes range from `0-5` to `1-5` for rating

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/helpers/feedback.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from app.models import db
+
+
+def delete_feedback(feedback):
+    """
+    Delete the feedback if rating is zero
+    :param feedback: Feedback to be deleted
+    :return:
+    """
+    db.session.delete(feedback)
```
Comment (mrsaicharan1) [inline app/api/helpers/feedback.py]: Just a question, Where are we checking if the feedback is 0?

Comment (shreyanshdwivedi) [reply app/api/helpers/feedback.py]: @mrsaicharan1 when rating is set to 0 on FE, `destroyRecord()` is called which sets `deleted_at` to non null value and then the rest is handled by the checks I introduced -
```
if feedback and 'deleted_at' in data and data['deleted_at']:
            if has_access('is_user_itself', user_id=feedback.user_id):
                delete_feedback(feedback)
            else:
                raise ForbiddenException({'source': ''},
                                         "Feedback can be deleted only by user himself")
```

File: app/api/feedbacks.py [hunk 2]
```diff
@@ -133,6 +134,12 @@ def before_update_object(self, feedback, data, view_kwargs):
             if session and not has_access('is_coorganizer', event_id=session.event_id):
                 raise ForbiddenException({'source': ''},
                                          "Event co-organizer access required")
+        if feedback and 'deleted_at' in data and data['deleted_at']:
```
Comment (iamareebjamal) [inline app/api/feedbacks.py]:
```suggestion
        if feedback and data.get('deleted_at'):
```

File: app/api/helpers/feedback.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    try:
+        db.session.commit()
+    except Exception as e:
+        logging.error('DB Exception! %s' % e)
```
Comment (iamareebjamal) [inline app/api/helpers/feedback.py]: Never ever use root logging, create a logger

Comment (iamareebjamal) [inline app/api/helpers/feedback.py]: logger.exception

File: app/api/schema/feedbacks.py [hunk 4]
```diff
@@ -23,7 +23,7 @@ class Meta:
         inflect = dasherize
 
     id = fields.Str(dump_only=True)
-    rating = fields.Float(required=True, validate=Range(min=0, max=5))
+    rating = fields.Float(required=True, validate=Range(min=1, max=5))
```
Comment (iamareebjamal) [inline app/api/schema/feedbacks.py]: This behaviour of frontend deleting the record when setting the rating to 0 is wrong IMO. Delete action should be very explicit. Either show a dialog on frontend that setting rating to 0 will delete the feedback

Comment (shreyanshdwivedi) [reply app/api/schema/feedbacks.py]: FE is using ui-rating from semantic UI. If a user selects rating 2 suppose and wants to clear out the rating, he/she needs to click on 2 stars again and the rating is cleared (set to 0). I think showing a dialog box will be better.
@CosmicCoder96 your opinion?

File: app/api/helpers/feedback.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    """
+    Delete the feedback if rating is zero
+    :param feedback: Feedback to be deleted
+    :return:
+    """
+    db.session.delete(feedback)
+    try:
+        db.session.commit()
+    except Exception as e:
+        logging.exception('DB Exception! %s' % e)
```
Comment (iamareebjamal) [inline app/api/helpers/feedback.py]: Still using root logging module.

Also, DB Exception is useless. It doesn't tell where the error happened.

Also, no need to include e in the log, stacktrace is automatically printed

Comment (shreyanshdwivedi) [reply app/api/helpers/feedback.py]: Ah, missed it. Fixed it now

File: app/api/schema/feedbacks.py (hunk 4 above)
Comment (shreyanshdwivedi) [reply app/api/schema/feedbacks.py]: @iamareebjamal please review this PR. The FE PR is still WIP so I can include your suggestions there

**Pull request merged**